addons/mail/tests/test_mail_template.py

```diff
@@ -46,6 +46,9 @@ def test_mail_compose_message_content_from_template_mass_mode(self):
         self.assertIn('13', values[self.partner_employee.id]['body_html'], 'We must trust mail template values')
 
     def test_mail_template_acl(self):
+        EXC_SELF_MSG = "Due to security restrictions, you are not allowed to modify 'Email Templates' (mail.template) records."
+        EXC_DYN_MSG = 'Only users belonging to the "Mail Template Editor" group can modify dynamic templates.'
```

Comment:
as discussed live, we could avoid the specific matching of exception by doing the steps in the right order.

The problem with the previous tests was that we could confuse an AccessError caused by the ownership of the template with an error caused by dynamic code being present. We can avoid that doubt by always having each "dynamic code edit" step preceded by a "static code edit" step *on the same template*, with *the same user*. With good comments to explain that this is important ;-)

That being said, I don't mind the extra specificity of the exception messages, it just makes the tests a bit more verbose/fragile.